zerver/lib/home.py

```diff
@@ -179,6 +196,8 @@ def build_page_params_for_home_page_load(
         # 2FA is not enabled.
         two_fa_enabled_user=two_fa_enabled and bool(default_device(user_profile)),
         is_web_public_visitor=user_profile is None,
+        # Disable events for web_public_visitors by tagging the queue as expired.
+        event_queue_expired=user_profile is None,
```

Comment:
Is this the right name for this feature?  I think `no_event_queue` would be more clear; the normal "expired" meaning for an event queue is that you need to reload to get one.

Reply:
I guess one thing to give some thought to is what the implications will be of not having event queues.  I suppose users will find that if they click around, new messages may appear, which is somewhat fundamental to the concept.  But we will have to deal with invariant failures like seeing a message containing a custom emoji that we don't have data on, etc.

I suppose experience will tell us if we need to add event queues for this (which is certainly doable, but may be a lot of work to finish nicely).

Reply:
Okay, noted!